  ** Applying Multiple Linear Regression to estimate the Medical Inurance Charges of an individual based on various factors provided in the Dataset.** 

---



---



*Importing Libraries*

---



In [0]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

*Importing the Dataset*

---



In [0]:
dataset=pd.read_csv("insurance.csv")
X = dataset.iloc[:, :-1].values
Y = dataset.iloc[:, -1:].values

*Printing the Dataset Values*

---



In [0]:
print(dataset.head())

In [0]:
print(X)

In [0]:
print(Y)

* Encoding Catagorical Data*

---



*Encoding Independent Factors - Gender,Smoker,Region *

In [0]:
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import LabelEncoder
le=LabelEncoder()
X[:,1]=(le.fit_transform(X[:,1]))
X[:,4]=(le.fit_transform(X[:,4]))
X[:,5]=(le.fit_transform(X[:,5]))


*Printing the Encoded Values from Dataset*

In [0]:
print(X)

*Splitting the Dataset into Training Set and Test Set *

---






In [0]:
from sklearn.model_selection import train_test_split
X_train,X_test,Y_train,Y_test = train_test_split(X,Y,test_size=0.3)

*Printing the Training Set Values*

---



In [0]:
print(X_train)

In [0]:
print(X_test)

*Printing the Test Values*

---



In [0]:
print(Y_train)

In [0]:
print(Y_test)

*Training Multiple Linear Regression On Training Set*

---



In [0]:
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,Y_train)

*Predicting the Values of the Test Set*

---



In [0]:
Y_pred=regressor.predict(X_test)

*Printing the Predicted Values*

---



In [0]:
np.set_printoptions(precision=2)
print(Y_pred)

*Concatenating the Predicted and Sample values to compare.*

---



In [0]:
print(np.concatenate((Y_pred,Y_test),1))

*Predicting a single value through inputs *

---



In [0]:
regressor.predict([[20,1,26.2,2,1,0]])

array([[27454.3]])

*Obtaining the Final Multiple Regression Eqaution. *

---



In [0]:
print(regressor.coef_)
print(regressor.intercept_)

[[  247.18   193.89   330.08   555.22 24366.09  -276.12]]
[-11807.82]


***The Final Equation Can be Represented as:***

---


***Charges = 247.18x age + 193.89x Sex + 330.08x bmi + 555.22x children + 24366.09xsmoker - 276.12x region - 11807.82 ***